src/wallet/rpcwallet.cpp

```diff
@@ -3299,6 +3449,9 @@ UniValue z_getbalance(const UniValue& params, bool fHelp)
     auto pa = keyIO.DecodePaymentAddress(fromaddress);
     fromTaddr = IsValidDestination(taddr);
     if (!fromTaddr) {
+        if (false /* Unified Address, see #5191 */) {
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid from address, cannot be a Unified Address.");
```

Comment:
It can't be a unified address because we want people to use `z_getbalanceforaddress` instead, so let's put that into the RPC error.